promsql/driver_collector.go

```diff
@@ -103,42 +57,52 @@ func NewDriverCollector(driver driver.Driver, opts DriverCollectorOpts) *DriverC
 	}
 
 	return &DriverCollector{
-		parent:                             driver,
-		DriverName:                         opts.DriverName,
-		QueryTotalCounter:                  prometheus.NewCounter(prometheus.CounterOpts{}),
-		QuerySuccessfulCounter:             prometheus.NewCounter(prometheus.CounterOpts{}),
-		QueryErroneousCounter:              prometheus.NewCounter(prometheus.CounterOpts{}),
-		TransactionTotalCounter:            prometheus.NewCounter(prometheus.CounterOpts{}),
-		TransactionSuccessfulAmountCounter: prometheus.NewCounter(prometheus.CounterOpts{}),
-		TransactionErroneousAmountCounter:  prometheus.NewCounter(prometheus.CounterOpts{}),
-		ExecutionTotalCounter:              prometheus.NewCounter(prometheus.CounterOpts{}),
-		ExecutionSuccessfulAmountCounter:   prometheus.NewCounter(prometheus.CounterOpts{}),
-		ExecutionErroneousAmountCounter:    prometheus.NewCounter(prometheus.CounterOpts{}),
+		parent:                       driver,
+		DriverName:                   opts.DriverName,
+		QueryTotalCounter:            prometheus.NewCounter(prometheus.CounterOpts{}),
+		QuerySuccessfulCounter:       prometheus.NewCounter(prometheus.CounterOpts{}),
+		QueryFailedCounter:           prometheus.NewCounter(prometheus.CounterOpts{}),
+		TransactionTotalCounter:      prometheus.NewCounter(prometheus.CounterOpts{}),
+		TransactionSuccessfulCounter: prometheus.NewCounter(prometheus.CounterOpts{}),
+		TransactionFailedCounter:     prometheus.NewCounter(prometheus.CounterOpts{}),
+		ExecutionTotalCounter:        prometheus.NewCounter(prometheus.CounterOpts{}),
+		ExecutionSuccessfulCounter:   prometheus.NewCounter(prometheus.CounterOpts{}),
+		ExecutionFailedCounter:       prometheus.NewCounter(prometheus.CounterOpts{}),
+
+		descQueryTotalCounter:            prometheus.NewDesc(fmt.Sprintf("db_%squery_total", prefix), "The total number of queries processed.", nil, nil),
```

Comment:
The `prometheus.CounterOpts` has this information about the name and "help" text. Use them instead of creating a `NewDesc`.